test/shopify-cli/task/ensure_env_test.rb

```diff
@@ -14,18 +14,152 @@ def setup
-        @task = EnsureEnv.call(@context)
-        assert_equal('test-shop.myshopify.com', Project.current.env.shop)
-      end
-
-      def test_writes_env_if_no_env
-        CLI::UI.expects(:ask).times(3)
-          .returns('apikey', 'apisecret', 'https://test-shop.myshopify.com')
-        @task = EnsureEnv.call(@context)
-        assert_equal('apikey', Project.current.env.api_key)
+      def test_create_new_app_if_none_available
+        response = [{
+          "id" => 421,
+          "businessName" => "one",
+          "stores" => [{
+            "shopDomain" => "store.myshopify.com",
+          }],
+          "apps" => [],
+        }]
+        ShopifyCli::PartnersAPI::Organizations.expects(:fetch_with_app).with(@context).returns(response)
+        CLI::UI::Prompt.expects(:ask).with(@context.message('core.tasks.ensure_env.app_name')).returns('new app')
+        CLI::UI::Prompt.expects(:ask).with(@context.message('core.tasks.ensure_env.app_type.select')).returns('public')
+        ShopifyCli::Tasks::CreateApiClient.expects(:call).with(
+          @context,
+          org_id: 421,
+          title: 'new app',
+          type: 'public',
+        ).returns({
+          "apiKey" => 1235,
+          "apiSecretKeys" => [{ "secret" => 1234 }],
+          "id" => "12345678",
+        })
+        Resources::EnvFile.expects(:new).with(new_env_file_values).returns(stub(:write))
+        assert_equal response.first, EnsureEnv.call(@context)
+      end
+
+      def test_no_prompt_if_one_app_and_org
+        response = [{
+          "id" => 421,
+          "businessName" => "one",
+          "stores" => [{
+            "shopDomain" => "store.myshopify.com",
+          }],
+          "apps" => [{
+            "title" => "app",
+            "apiKey" => 1235,
+            "apiSecretKeys" => [{
+              "secret" => 1234,
+            }],
+          }],
+        }]
+        ShopifyCli::PartnersAPI::Organizations.expects(:fetch_with_app).with(@context).returns(response)
+        Resources::EnvFile.expects(:new).with(new_env_file_values).returns(stub(:write))
+        assert_equal response.first, EnsureEnv.call(@context)
+      end
+
+      def test_uses_default_values_for_env_file
+        Resources::EnvFile.expects(:parse_external_env).raises(Errno::ENOENT)
+        ShopifyCli::PartnersAPI::Organizations.expects(:fetch_with_app).with(@context).returns(partners_api_response)
+        expect_user_prompts
+        Resources::EnvFile.expects(:new).with(new_env_file_values).returns(stub(:write))
+        assert_equal partners_api_response[1], EnsureEnv.call(@context)
+      end
+
+      def test_keep_existing_env_values
+        Resources::EnvFile.expects(:parse_external_env).returns({ host: "host" })
+        ShopifyCli::PartnersAPI::Organizations.expects(:fetch_with_app).with(@context).returns(partners_api_response)
+        expect_user_prompts
+        Resources::EnvFile.expects(:new).with(new_env_file_values.merge({ host: "host" })).returns(stub(:write))
```

Comment:
I don't think I was clear with my previous comment.  I think we should use `expects` instead of `stub` here.  In case you want to read up more about mocks vs stubs, [here](https://www.rubyguides.com/2018/10/rspec-mocks/) is a good explanation. TLDR:  "A stub is only a method with a canned response, it doesn’t care about behavior.  A mock expects methods to be called, if they are not called the test will fail."  I would only use `stub` if I don't care if that method is called or not.  In this case, I think we do want to verify that the `.write` is called by using `expects`.  So I suggest something like this:
```
    env_file = Minitest::Mock.new
    Resources::EnvFile.expects(:new).with(...).returns(env_file)
    env_file.expects(:write).once
```